six/modules/c++/six.sicd/tests/test_load_from_input_stream.cpp

```diff
@@ -0,0 +1,120 @@
+/* =========================================================================
+* This file is part of six-c++
+* =========================================================================
+*
+* (C) Copyright 2004 - 2016, MDA Information Systems LLC
+*
+* six-c++ is free software; you can redistribute it and/or modify
+* it under the terms of the GNU Lesser General Public License as published by
+* the Free Software Foundation; either version 3 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public
+* License along with this program; If not,
+* see <http://www.gnu.org/licenses/>.
+*
+*/
+
+#include <iostream>
+#include <stdexcept>
+#include <memory>
+#include <complex>
+
+#include <sys/Path.h>
+#include <except/Exception.h>
+#include <io/StandardStreams.h>
+#include <six/sicd/ComplexXMLControl.h>
+#include <six/sicd/Utilities.h>
+#include <six/sicd/ComplexData.h>
+#include <nitf/IOStreamReader.hpp>
+
+int main(int argc, char** argv)
+{
+    try
+    {
+        // Parse the command line
+        const std::string progname(argv[0]);
+        if (argc != 2 && argc != 3)
+        {
+            std::cerr << "Usage: " << sys::Path::basename(progname)
+                << " <SICD pathname> [<schema dirname>]\n\n";
+            return 1;
+        }
+
+        const std::string sicdPathname(argv[1]);
+
+        std::vector<std::string> schemaPaths;
+        if (argc == 3)
+        {
+            schemaPaths.push_back(argv[2]);
+        }
+        else
+        {
+            // In a normal installation, we can infer the path
+            const sys::Path progDirname =
```

Comment:
Can you make a utility function for this somewhere common that we can use for programs like this?